Graph-iOS-Swift-Snippets/AuthenticationProvider.swift

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license.
+ * See LICENSE in the project root for license information.
+ */
+
+import Foundation
+import MSAL
+import MSGraphSDK
+
+class AuthenticationProvider: NSObject, MSAuthenticationProvider
+{
+    var accessToken: String?
+    var account: MSALAccount?
+    var msalClient: MSALPublicClientApplication
+    
+    init(clientId: String, authority: MSALAuthority) throws
+    {
+        let config = MSALPublicClientApplicationConfig(clientId: clientId)
+        msalClient = try MSALPublicClientApplication(configuration: config)
+    }
+    
+    func acquireAuthToken(scopes: [String], completion:@escaping (_ success:Bool, _ error: NSError?) -> Void)
+    {
+        var completionBlock: MSALCompletionBlock?
+        completionBlock = { (result, error) in
+            
+            if let error = error as NSError?
+            {
+                if (error.domain == MSALErrorDomain)
+                {
+                    let msalError = MSALError(rawValue: error.code)!
```

Comment:
why do we need to create MSALError?

Reply:
We use it in switch below. We just map integer to enum here.